src/Opaleye/Internal/PrimQuery.hs

```diff
@@ -53,6 +53,7 @@ data PrimQuery' a = Unit
                               (Bindings HPQ.PrimExpr)
                               (PrimQuery' a)
                               (PrimQuery' a)
+                  | Antijoin  (PrimQuery' a) (PrimQuery' a)
```

Comment:
Oh, now I see that there is no such join type as `ANTIJOIN`, but rather "antijoin" is the name for the pattern "WHERE NOT EXISTS".

No need to stick it in `Join`/`JoinType` then.  It almost certainly won't work.